diagnostic.md

```diff
@@ -100,11 +131,24 @@ object.
 ```
 
 ```sh
-# your answer here
+curl --include --request PATCH http://www.example.com/donuts/42 \
+--header "Content-Type: application/json" \
+--data '{
+  "donuts": {
+    "name": "Krüller"
+  }
+}'
+
 ```
 
 Write an AJAX request to change the donut on the server using JSON.
 
 ```js
-let changeDonut = /* your answer here */;
+let changeDonut = function({"name": "Krüller"}){
+  return $.ajax({
+    url: http://www.example.com/donuts/42,
+    method: 'PATCH',
+    data,
+  });
+}
```

Comment:
you need to hardcode the data like this:

```js
let changeDonut = function (donutID) {
  return $.ajax({
    url: 'http://example.com/donuts/' + donutID,
    method: 'PATCH',
    data: {
      "donut" {
        "name": "Krüller",
      }
    }
  });
};
```

I can see what you're trying to do, giving the data you want to the functions arguments but then the data you use on line 151 is undefined.